### PR TITLE
DT-2070: ⬆️  Upgrade to spring boot 2.5.0 and kotlin 1.5.10

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/functional/GradleBuildTest.kt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 package uk.gov.justice.digital.hmpps.gradle.functional
 
-import org.gradle.internal.impldep.org.codehaus.plexus.util.FileUtils
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/functional/GradleBuildTest.kt:3]: strange import corrected

File: src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/functional/ProjectRunner.kt [hunk 2]
```diff
@@ -19,13 +19,16 @@ private const val DEBUG_TESTS = false // To debug functional tests change this t
 data class ProjectDetails(
   val projectDir: File,
   val projectName: String,
-  val packageDir: String,
+  val packageDir: String = "src/main/java/uk/gov/justice/digital/hmpps/app",
+  val resourcesDir: String = "src/main/resources",
   val mainClassName: String,
   val mainClass: String,
+  val applicationYml: String = "application.yml",
+  val applicationScript: String = "management.endpoints.web.exposure.include: health,info",
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/functional/ProjectRunner.kt:27]: need to make info public for all tests to pass

Comment (petergphillips) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/functional/ProjectRunner.kt:27]: Raised https://dsdmoj.atlassian.net/browse/DT-2071 to see if we need to protect that endpoint instead

File: release-notes/3.3.0.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# 3.3.0
+
+## Upgrade to Spring Boot 2.5
+
+Since this is a minor Spring Boot release there are [upgrade instructions](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.5-Release-Notes#upgrading-from-spring-boot-24) to follow.  `spring.datasource` has been renamed to `spring.sql.init` and `/info` is now protected by default.
+
+To expose `/info` to public add
+```
+management.endpoints.web.exposure.include: health,info
```
Comment (mikehalmamoj) [inline release-notes/3.3.0.md:9]: Is this a breaking change?  If so maybe consider going to version 4.0.0 when you have tested the beta?

**Review by mikehalmamoj — APPROVED** [review]

File: release-notes/3.3.0.md (hunk 3 above)
Comment (petergphillips) [reply release-notes/3.3.0.md:9]: Difficult one.  I did think about this, but Spring Boot only went from 2.4 to 2.5, so was following the same convention

Comment (mikehalmamoj) [reply release-notes/3.3.0.md:9]: Yes, but Spring don't seem to follow the rules of semver anymore but we claim to in our readme :smile: .  Personally I'd go for a major version but probably needs a team decision.

**Pull request merged**